homeassistant/components/rainbird/__init__.py

```diff
@@ -31,24 +32,31 @@
 SENSOR_TYPE_RAINSENSOR = "rainsensor"
 
 
-class RainBirdSensorMetadata(NamedTuple):
-    """Metadata for an individual RainBird sensor."""
-
-    name: str
-    icon: str
-    unit_of_measurement: str | None = None
-
+SENSOR_TYPES: tuple[SensorEntityDescription, ...] = (
+    SensorEntityDescription(
+        key=SENSOR_TYPE_RAINSENSOR,
+        name="Rainsensor",
+        icon="mdi:water",
+    ),
+    SensorEntityDescription(
+        key=SENSOR_TYPE_RAINDELAY,
+        name="Raindelay",
+        icon="mdi:water-off",
+    ),
+)
 
-SENSOR_TYPES: dict[str, RainBirdSensorMetadata] = {
-    SENSOR_TYPE_RAINSENSOR: RainBirdSensorMetadata(
-        "Rainsensor",
+BINARY_SENSOR_TYPES: tuple[BinarySensorEntityDescription, ...] = (
+    BinarySensorEntityDescription(
+        key=SENSOR_TYPE_RAINSENSOR,
+        name="Rainsensor",
         icon="mdi:water",
     ),
```

Comment:
I don't quite like that I had to duplicate these: once for `Sensors` and once for `BinarySensors`. @frenck Do you have an idea how to improve it and still be compatible with typing?

Reply:
Hmm... I would say this is more of an edge case. There is no real good way to do that.

Looking at it, I wonder why it adds them both for the same data even... IMHO that is that part where it goes wrong :)

Reply:
In that case, let's keep it as is for now. A major refactoring with a breaking changing isn't the goal with this PR